### PR TITLE
Implement debug framework

### PR DESCRIPTION
**Pull request by liamawhite**
This also enables a basic admin API collection. I will follow up this PR with tarballing and tests for admin API collection in order to keep the size of this PR down.

Resolves https://github.com/tetratelabs/getenvoy/issues/30

Signed-off-by: Liam White <liam@tetrate.io>

### CODE REVIEW

File: pkg/binary/getenvoy/termination.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	"stats?format=json": "stats.json",
+	"runtime":           "runtime.json",
+}
+
+// EnableEnvoyAdminDataCollection registers collection of Envoy Admin API information
+var EnableEnvoyAdminDataCollection = func(r *Runtime) {
+	r.registerPreTermination(func(r *Runtime) error {
+		var multiErr *multierror.Error
+		for path, file := range adminAPIPaths {
+			resp, err := http.Get(fmt.Sprintf("http://0.0.0.0:15001/%v", path))
```
Comment (lizan) [inline pkg/binary/getenvoy/termination.go]: localhost?

File: pkg/binary/getenvoy/termination.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	// Execute all registered preTermination functions
+	for _, f := range r.preTermination {
+		if err := f(r); err != nil {
+			log.Error(err.Error())
+		}
+	}
+
+	// Forward on the SIGINT to Envoy
+	log.Infof("Sending Envoy process (PID=%d) SIGINT", r.cmd.Process.Pid)
+	_ = r.cmd.Process.Signal(syscall.SIGKILL)
```
Comment (lizan) [inline pkg/binary/getenvoy/termination.go]: SIGINT?

Comment (liamawhite) [reply pkg/binary/getenvoy/termination.go]: Good catch, changed for some manual testing. Will change back.

**Comment by liamawhite** [comment]
Merging to unblock incoming PRs as already approved.

**Pull request merged**